packages/shared/lib/core/ledger/utils/openLedgerConfirmationPopup.ts

```diff
@@ -0,0 +1,37 @@
+import { PopupProps } from '@auxiliary/popup'
+import { isPreparedTransaction, isPreparedTransactionEssenceHash } from '@core/profile-manager'
+import { TransactionProgressEventPayload } from '@core/profile-manager/api/types'
+import { openPopup } from '@lib/popup'
+
+import { deconstructLedgerSendConfirmationProps } from '../helpers'
+
+export function openLedgerConfirmationPopup(
+    payload?: TransactionProgressEventPayload,
+    isDuringOnboarding?: boolean
+): void {
+    if (!payload) {
+        return
+    } else {
+        let props: PopupProps
+        switch (true) {
+            case isPreparedTransaction(payload):
+                props = {
+                    ...deconstructLedgerSendConfirmationProps(),
+                    needsToShowPopupAfterwards: !isDuringOnboarding,
+                }
+                break
+            case isPreparedTransactionEssenceHash(payload):
+                props = { hash: payload?.['PreparedTransactionEssenceHash'] }
```

Comment:
Here optional chaining is not required, since you already check whether payload contains the `PreparedTransactionEssenceHash` in l.23

Reply:
True, just an automatic habit for me :sweat_smile: